test/features/sync.test.js

```diff
@@ -55,12 +54,55 @@ describe("Sync", () => {
         });
       }
     }));
-
+    console.log("!!!c");
     await background.browser.runtime.onStartup.addListener.yield();
     await nextTick();
 
     const sync = await background.browser.storage.sync.get();
+    console.log(await background.browser.storage.local.get());
 
     expect(sync.identities.length).to.equal(5);
+    console.log("!!!b");
+  });
+
+  it("should sync for the first time", async () => {
```

Comment:
Just curious, I see you started working with the tests, but seem to have dropped it again and reverted back to testing directly in the browser - was there something not working as expected or had it a specific reason?

Reply:
Yes, I was having a really hard time with the test suite. I've never used Mocha, jsdom, etc before. I'm sure I could figure it out, but I didn't want to sink all of my time into it. There are tests in here, and I'll discuss with @groovecoder how to move forward from here.

Reply:
I think I can do most of what I'm doing in `test.js` with `background.browser.runtime.onMessage` (your suggestion) to actually get it into the test suite, but that seemed a little 'round-about'. That's probably the minimum I would do. I bet there's better ways to handle this, though. 

Do you know if we can do unit tests on a given function with mocha if this is not a node application? Everything I've seen makes it look pretty easy if the files are modules, but obviously that breaks the add on (oh, I tried it!). We discussed browserifying this next if that makes it easier/possible to do unit tests in Mocha.

Reply:
Ah, yeah, testing with new tooling can be tricky and time-consuming indeed. The tests file for sync looks good tho, could possibly get merged into the actual tests afterwards.

> Do you know if we can do unit tests on a given function with mocha if this is not a node application?

The classes/functions could be exposed on `window` and then accessed with `background.window` in the tests. Not exactly pretty, but possibly easier than introducing bundling with webpack or similar.

If you're going to give testing with mocha another try, I'd be happy to help if I can, you could reach me on matrix.org (the chat platform Mozilla is moving to) at https://matrix.to/#/@stoically:matrix.org